x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java

```diff
@@ -403,6 +405,7 @@ protected void onFetchFailure(int shardIndex, SearchShardTarget shardTarget, Exc
         protected void onListShards(List<SearchShard> shards, List<SearchShard> skipped, Clusters clusters, boolean fetchPhase) {
             // best effort to cancel expired tasks
             checkCancellation();
+            ccsMinimizeRoundtrips = clusters instanceof SearchResponse.CcsClusters ccsClusters && ccsClusters.isCcsMinimizeRoundtrips();
```

Comment:
you can remove the instanceof checks by adding the isCcsMinimizeRoundtrips method to the base class and make it always return false?